Automattic/wp-calypso
# Playwright: separate editor-related methods into `EditorGutenbergComponent`.

#### Changes proposed in this Pull Request

This PR removes editor-related methods from `GutenbergEditorPage`, allowing the rename of said file into `EditorPage`.

Key changes:
- extraction of editor and block-related methods into new child class `EditorGutenbergComponent`.
- removal of dead selectors.
- rename `GutenbergEditorPage` to `EditorPage` in order to decouple the page object from its implementation.

#### Testing instructions

Ensure the following:
  - [x] i18n E2E

Related to #60868
Depends on https://github.com/Automattic/wp-calypso/pull/61669.
